### PR TITLE
Mark file server ingests private

### PR DESCRIPTION
**Pull request by lfarrell**
https://jira.lib.unc.edu/browse/BXC-2001

### CODE REVIEW

File: deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/StaffOnlyPermissionJobTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    private final static String LOC1_ID = "loc1";
+    private final static String LOC2_ID = "loc2";
+    private final static String FILE_CONTENT1 = "Some content";
+    private final static String SOURCE_ID = "source1";
+
+    private StaffOnlyPermissionJob job;
+    private Model model;
+    private StorageLocationManager locationManager;
+    private Bag depBag;
+    private StorageLocationTestHelper locTestHelper;
```
Comment (bbpennel) [inline deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/StaffOnlyPermissionJobTest.java]: you don't need any of the storage location related stuff in this test since the job doesn't interact with the binaries. So you can remove all the variables below this line, and the constants that go with them.

File: deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/StaffOnlyPermissionJobTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        Map<String, String> depositStatus = new HashMap<>();
+        depositStatus.put(DepositField.staffOnly.name(), "true");
+        when(depositStatusFactory.get(eq(depositUUID))).thenReturn(depositStatus);
+
+        PID objPid = makePid();
+        Bag objBag = model.createBag(objPid.getRepositoryPath());
+        objBag.addProperty(RDF.type, Cdr.Folder);
+
+        Resource fileResc = addFileObject(objBag, FILE_CONTENT1);
+        objBag.addProperty(Cdr.primaryObject, fileResc);
```
Comment (bbpennel) [inline deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/StaffOnlyPermissionJobTest.java]: i know this is test data, but a folder isn't allow to have a primaryObject. You could remove this property, or make objBag into a work. Also, the child could be any type, it being a file with a storage location isn't necessary for this test.

File: metadata/src/main/java/edu/unc/lib/dl/rdf/Cdr.java [hunk 3]
```diff
@@ -140,6 +140,13 @@ public static String getURI() {
      */
     public static final Property replaceInvalidTerms = createProperty(
             "http://cdr.unc.edu/definitions/model#replaceInvalidTerms" );
+
+    /**
+     * Relation which denotes the permissions for the subject object
+     */
+    public static final Property staffOnly = createProperty(
```
Comment (bbpennel) [inline metadata/src/main/java/edu/unc/lib/dl/rdf/Cdr.java]: Revert the changes in this file, this only needs to be a deposit status field

File: metadata/src/main/java/edu/unc/lib/dl/rdf/CdrDeposit.java [hunk 4]
```diff
@@ -110,4 +110,7 @@ public static String getURI() {
 
     public static final Property hasSourceMetadata = createProperty(
             "http://cdr.unc.edu/definitions/deposit#hasSourceMetadata");
+
+    public static final Property staffOnly = createProperty(
```
Comment (bbpennel) [inline metadata/src/main/java/edu/unc/lib/dl/rdf/CdrDeposit.java]: Revert the changes in this file, this only needs to be a deposit status field

File: static/js/admin/src/IngestFromSourceForm.js [hunk 5]
```diff
@@ -139,10 +154,12 @@ define('IngestFromSourceForm', [ 'jquery', 'AbstractFileUploadForm', 'ModalLoadi
 			$this.parents(".file_browse_heading").first().nextUntil(".file_browse_heading", ".file_browse_entry").each(function() {
 				if (select) {
 					$(this).addClass("selected");
+					$("input[name=ingest]").prop("checked", select);
 				} else {
 					$(this).removeClass("selected");
+					$("input").prop("checked", select);
```
Comment (bbpennel) [inline static/js/admin/src/IngestFromSourceForm.js]: Won't this selector find all the input elements on the page now?

File: static/templates/admin/ingestSourceMetadata.html [hunk 6]
```diff
@@ -3,51 +3,51 @@
...
+
 	<div class="file_browse_list">
-	<table>
-	<%
-	_.each(selectedCandidates, function(candidate, index) { 
-		%>
-		<tr class="file_browse_entry">
-			<td class="file_browse_type_icon">
-				<% if (candidate.type == "BagIt") { %>
+		<table>
```
Comment (bbpennel) [inline static/templates/admin/ingestSourceMetadata.html]: Were all the changes in this file just formatting changes? I'm not actually seeing a checkbox for making something private here or when i run the branch

File: static/js/admin/src/IngestFromSourceForm.js (hunk 5 above)
Comment (lfarrell) [reply static/js/admin/src/IngestFromSourceForm.js]: No, it just finds all the checkboxes in the selected row

File: deposit/src/main/java/edu/unc/lib/deposit/fcrepo4/IngestDepositRecordJob.java [hunk 7]
```diff
@@ -145,6 +145,10 @@ private Resource makeDepositRecord(Resource deposit, Map<String, String> status)
         if (onBehalfOf != null) {
             aipObjResc.addProperty(Cdr.depositedOnBehalfOf, onBehalfOf);
         }
+        String staffOnly = status.get(DepositField.staffOnly.name());
```
[the remark this answers is not in the document]
Comment (lfarrell) [reply deposit/src/main/java/edu/unc/lib/deposit/fcrepo4/IngestDepositRecordJob.java]: This should be used by the line right below it.

[the remark this answers is not in the document]
Comment (lfarrell) [reply deposit/src/main/java/edu/unc/lib/deposit/fcrepo4/IngestDepositRecordJob.java]: Never mind, I see what you mean.

File: static/js/admin/src/IngestFromSourceForm.js (hunk 5 above)
Comment (bbpennel) [reply static/js/admin/src/IngestFromSourceForm.js]: maybe i'm misunderstanding, but i thought jquery would use the whole document as the context for a lookup if you don't specify a context. Is it somehow inheriting a context?

File: deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/StaffOnlyPermissionJobTest.java [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+        model = job.getReadOnlyModel();
+        Resource roles = model.getResource(objPid.getRepositoryPath());
+
+        assertTrue(roles.hasProperty(none.getProperty(), AUTHENTICATED_PRINC));
+        assertTrue(roles.hasProperty(none.getProperty(), PUBLIC_PRINC));
+
+        assertFalse(fileResc.hasProperty(none.getProperty()));
+
+        originalFile.deleteOnExit();
```
Comment (bbpennel) [inline deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/StaffOnlyPermissionJobTest.java]: You can do this immediately after you create the follow rather than in each test. Although i'll mention that again that the job doesn't do anything with files, so you don't really need to have actual files and staging locations set up

File: static/templates/admin/ingestSourceMetadata.html [hunk 9]
```diff
@@ -35,11 +35,13 @@ <h2><%= candidate.patternMatched %></h2>
...
 					<label for="file_label">Label <span class="required">*</span></label>
-					<input type="text" name="file_label" size="40" value="<%= candidate.filename %>" />
+					<input id="file_label" type="text" name="file_label" size="40" value="<%= candidate.filename %>" />
 					<label for="file_acc_number">Accession Number</label>
-					<input type="text" name="file_acc_number" size="40"/>
+					<input id="file_acc_number" type="text" name="file_acc_number" size="40"/>
 					<label for="file_media_id">Media ID</label>
-					<input type="text" name="file_media_id" size="40"/>
+					<input id="file_media_id" type="text" name="file_media_id" size="40"/>
+					<label for="staff_only">Staff only</label>
```
Comment (bbpennel) [inline static/templates/admin/ingestSourceMetadata.html]: The ticket suggests using the label `Restrict to staff only access?`

File: deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/StaffOnlyPermissionJobTest.java (hunk 8 above)
Comment (lfarrell) [reply deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/StaffOnlyPermissionJobTest.java]: No the badge branch isn't in here
